src/sizzle.js

```diff
@@ -128,7 +128,18 @@ var i,
 	rnative = /^[^{]+\{\s*\[native \w/,
 
 	// Easily-parseable/retrievable ID or TAG or CLASS selectors
-	rquickExpr = /^(?:#([\w-]+)|(\w+)|\.([\w-]+))$/,
+	// ID: #((?:[\w-]|\\[\!\"\#\$\%\&\'\(\)\*\+\,\-\.\/\:\;\<\=\>\?\@\\\[\]\^\`\{\|\}\~]|\\[a-f0-9]{1,6})+)
```

Comment:
A regexp this complex would need to be thoroughly tested in the unit tests, and there are no additional tests here. I also don't see any perf test.

Reply:
I can add it. 
I ran on my own website and it has factor of 1000 in IE.